util/ci-run-file-creator.py

```diff
@@ -222,6 +222,35 @@ def update_userenvs():
                     else:
                         log.info("Creating %s userenv for endpoint %d since requested userenv is not default" % (args.endpoint, endpoint_idx))
                         endpoint["userenv"] = args.userenv
+        case "remotehosts":
+            endpoint = run_file["endpoints"][0]
+            if "userenv" in endpoint["settings"]:
+                if args.userenv == "default":
+                    log.info("Found existing %s global userenv but since requested userenv is default is is being removed" % (args.endpoint))
+                    del endpoint["settings"]["userenv"]
+                else:
+                    log.info("Updating existing %s global userenv" % (args.endpoint))
+                    endpoint["settings"]["userenv"] = args.userenv
+            else:
+                if args.userenv == "default":
+                    log.info("No %s global userenv to update since requested userenv is default" % (args.endpoint))
+                else:
+                    log.info("Creating %s global userenv since requested userenv is not default" % (args.endpoint))
+                    endpoint["settings"]["userenv"] = args.userenv
```

Comment:
if not default and not defined in the json, where this args.userenv comes from ?

Reply:
The invocation of the script by `crucible-ci`:

https://github.com/perftool-incubator/crucible-ci/blob/be052c3b92a2e15fb8ed2f5a8e13e542094e697d/.github/actions/integration-tests/run-ci-stage1#L534-L578